utils/submissionDataFromGoogleMaps.ts

```diff
@@ -0,0 +1,124 @@
+import puppeteer from 'puppeteer'
+import dotenv from 'dotenv'
+
+//Load the api key for google maps
+dotenv.config()
+
+// This function ensures that we can redirect in a headerless browser using puppeteer to extract url once it has coordinates
+const extractRedirectedUrl = async (initialUrl: string): Promise<string> => {
+  const browser = await puppeteer.launch({ headless: true }) // ensures it doesn't open an actual browser
+
+  try {
+    const page = await browser.newPage()
+    await page.goto(initialUrl, {
+      waitUntil: 'networkidle0', // Ensures that the network page has loaded to have coordinates
+      timeout: 120000, // long timeout so that it has time to redirect. Google redirects twice from mobile links
+    })
+    return page.url()
+  } catch (error) {
+    console.error('Error extracting redirected URL:', error)
+    throw error
+  } finally {
+    await browser.close()
+  }
+}
+
+// Extract the coordinates from the extracted google maps url
+const extractCoordinatesFromRedirectedUrl = (url: string): { latitude: string, longitude: string } | null => {
+  // regex to extract the full coordinates from url
+  const regex = /@(-?\d+\.\d+),(-?\d+\.\d+),/
+  const match = url.match(regex)
+
+  if (match) {
+    const latitude = match[1]
+    const longitude = match[2]
+    return { latitude, longitude }
+  } else {
+    console.error('No coordinates found in the URL.')
+    return null
+  }
+}
+
+
+
+// This function uses the places api to get the closest clinic to the coordinates in the url
+const getNearbyLocationsFromCoordinates = async (latitude: string, longitude: string, apiKey: string): Promise<any | null> => {
+  const url = 'https://places.googleapis.com/v1/places:searchText'
```

Comment:
What I have read so far is that you can fetch the place id by using the coordinates. So this could be another possibility, fetch the place id by coordinates, than fetch the location details using the place id. Not sure if this is something you already tried or did

Reply:
The problem with that solution is that in Japan. Tall buildings can have many clinics and they are at the same coordinates

Reply:
Hmm yeah I see. Maybe LaShawn's proposal is maybe a better way than by using a phone number, since than we are 100% sure to get the write place if we assume that the user gives up the right phone number. 

How do we make sure now that we get the right place?

Reply:
We don't know we get the right place all the time unfortunately. That is really the moderators job to double check. This is just to help with filling the form and less work if what they confirmed is correct right away